src/board/devices/output/7seg.ts

```diff
@@ -0,0 +1,205 @@
+    [false, false, false, false, false, false, false, false],
+    [false, false, false, false, false, false, false, false],
+    [false, false, false, false, false, false, false, false],
+    [false, false, false, false, false, false, false, false]
+  ]
+  /**
+   * Returns true if seg with given position is on.
+   *
+   * @param position seg position to check (0-31)
+   * @param segment segmentnumber to check (0-3)
+   * @returns true if segment is turned on
+   */
+  public isOn(segment: number, position: number): boolean {
+    if (SEVENseg.invalidPosition(segment, position)) {
+      throw new Error(`Segment ${segment} Position ${position} does not exist.`)
+    }
+    return this.displays[segment][position]
+  }
+
+  /**
+   * Returns an array of segment positions,true if seg on position should be on.
+   * the values are read from the newer updated datasection and interpreted accordingly
+   *
+   * @param display display to get (0-3)
+   * @returns segmentArray an array of boolean, true if segment is turned on
+   */
+  public getDisplay(display: number): boolean[] {
+    if (this.isBin()) {
+      this.displays[display] = this.getBinDisplay(display)
+    }
+    if (this.isReg()) {
+      this.displays[display] = this.getRegDisplay(display)
+    }
+    return this.displays[display]
+  }
+  /**
+   * Returns true if value where the binary section is stored has changed.
+   *
+   * @returns true in case of change
+   */
+  private isBin(): boolean {
+    return (
+      this.memory.readHalfword(this.startAddressBin).toUnsignedInteger() !=
+      this.oldBin.toUnsignedInteger()
+    )
+  }
+  /**
+   * Returns true if value where the regular section is stored has changed.
+   *
+   * @returns true in case of change
+   */
+  private isReg(): boolean {
+    return (
+      this.memory.readWord(this.startAddress).toUnsignedInteger() !=
+      this.oldSeg.toUnsignedInteger()
+    )
+  }
+  /**
+   * converts an number to an array of segment positions,true if seg on position should be on.
+   *
+   * @param inNum number to convert (0-15)
+   * @returns segmentArray an array of boolean, true if segment is turned on
+   */
+  private toSevensegNrBin(inNum: number): boolean[] {
+    let arr: boolean[] = []
+
+    let code: String = SEVENsegNr[inNum]
+    for (let i = 0; i < code.length; i++) {
+      const character = code.charAt(i)
+      if (character == '0') {
+        arr.push(false)
+      }
+      if (character == '1') {
+        arr.push(true)
+      }
+      if (character == '_') {
+      }
+    }
+
+    return arr
+  }
+
+  /**
+   * Returns an array of segment positions,true if seg on position should be on.
+   * the values are read from the datasection and interpreted binary
+   *
+   * @param display to check (0-3)
+   * @returns segmentArray an array of boolean, true if segment is turned on
+   */
+  private getBinDisplay(display: number): boolean[] {
+    let arr: boolean[] = []
+    switch (display) {
```

Comment:
Here instead of working with your enum above, you could just read the memory at either address `0x60000114` for display number 0-1 or at address `0x60000115` at display number 2-3.

Afterwards, depending on the display number, you need the lower or the higher nibble of the byte. So you could fetch the unsigned value of the byte that was read from the address and do a `AND` operation with the mask `0xF` for the lower nibble and `0xF0` for the higher nibble. For the higher nibble you would also need to do `>> 4`. Now you should get a number between 0..15.

Now instead of using the enum with the strings I would have created a Map that has as key the number 0...15 and as value your boolean array that indicates which positions should be set to true or false in the display.